### PR TITLE
SW-3099 Export/download inventory batches from the seedlings table

### PR DESCRIPTION
**Pull request by karthikbtf**
- reused modal from accessions download (refactored it for reuse)
- verified the download works as required by the jira ticket, result csv attached

Terraware App 2023-05-03 15-45-07

Terraware App 2023-05-03 15-44-49

[test_ggggaaa_facilities.csv](https://github.com/terraware/terraware-web/files/11391773/test_ggggaaa_facilities.csv)

### CODE REVIEW

File: src/components/Inventory/view/InventorySeedlingsTable.tsx [hunk 1]
```diff
@@ -58,33 +61,33 @@ export default function InventorySeedslingsTable(props: InventorySeedslingsTable
     { key: 'quantitiesMenu', name: '', type: 'string' },
   ];
 
-  useEffect(() => {
-    let activeRequests = true;
+  const getSearchFields = useCallback(() => {
```
Comment (karthikbtf) [inline src/components/Inventory/view/InventorySeedlingsTable.tsx:64]: this was extracted out into a callback, no change in code

File: src/components/Inventory/view/InventorySeedlingsTable.tsx [hunk 2]
```diff
@@ -209,87 +212,111 @@ export default function InventorySeedslingsTable(props: InventorySeedslingsTable
...
+              controlledOnSelect={true}
+              sortHandler={onSortChange}
+              isPresorted={!!searchSortOrder}
+            />
+          </Box>
+        </Grid>
       </Grid>
-    </Grid>
+    </>
   );
```
Comment (karthikbtf) [inline src/components/Inventory/view/InventorySeedlingsTable.tsx:321]: all of this was just indented right by 2 spaces

File: src/components/common/ExportCsvModal.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import { Grid } from '@mui/material';
+import TextField from 'src/components/common/Textfield/Textfield';
+
+interface ExportCsvModalProps {
+  open: boolean;
+  onExport: () => Promise<any>;
+  onClose: () => void;
+}
+
+export default function ExportCsvModal(props: ExportCsvModalProps): JSX.Element {
```
Comment (karthikbtf) [inline src/components/common/ExportCsvModal.tsx:14]: extracted out from accessions download report modal

File: src/components/Inventory/view/BatchesExportModal.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+interface DownloadReportModalProps {
+  organizationId: number;
+  speciesId: number;
+  searchFields: SearchNodePayload[];
+  searchSortOrder?: SearchSortOrder;
+  onClose: () => void;
+}
+
+export default function DownloadReportModal(props: DownloadReportModalProps): JSX.Element {
```
Comment (constanzauanini) [inline src/components/Inventory/view/BatchesExportModal.tsx:13]: Should we call the component BatchesExportModal? And BatchesExportModalProps ?

**Review by constanzauanini — APPROVED** [review]

File: src/components/Inventory/view/BatchesExportModal.tsx (hunk 4 above)
Comment (karthikbtf) [reply src/components/Inventory/view/BatchesExportModal.tsx:13]: Yes, we should - thanks :)
Will fix it.

**Comment by karthikbtf** [comment]
I will get this in *after* the production release - so Allison has a chance to QA it before it goes to customer.

**Pull request merged**